test/error_test.exs

```diff
@@ -0,0 +1,17 @@
+defmodule ErrorTest do
+  use ExUnit.Case, async: true
+
+  alias Postgrex, as: P
+
+  @tag min_pg_version: "9.3"
+  test "connection works after failure in executing state" do
```

Comment:
this is great addition but we should rename this test and maybe move it to query_test.exs.